rpcs3/rpcs3qt/log_frame.cpp

```diff
@@ -430,25 +430,26 @@ void log_frame::UpdateUI()
 {
 	const auto start = steady_clock::now();
 
+	const auto tty = g_fxo->get<tty_t>();
+
 	// Check TTY logs
-	while (const u64 size = std::max<s64>(0, g_tty_size.load() - (m_tty_file ? m_tty_file.pos() : 0)))
+	if (tty && tty->fd && tty->fd.size() && m_tty_size != tty->fd.size())
 	{
-		std::string buf;
-		buf.resize(size);
-		buf.resize(m_tty_file.read(&buf.front(), buf.size()));
-
-		if (buf.find_first_of('\0') != umax)
+		if (m_TTYAct->isChecked())
 		{
-			m_tty_file.seek(s64{0} - buf.size(), fs::seek_mode::seek_cur);
-			break;
-		}
+			std::string buf;
+			{
+				std::lock_guard lock(tty->mtx);
+				const auto pos = tty->fd.pos();
+				tty->fd.seek(m_tty_size);
+				tty->fd.read(buf, tty->fd.size() - m_tty_size);
+				tty->fd.seek(pos);
+				m_tty_size = tty->fd.size();
```

Comment:
Well. As I mentioned before, now you'll have the problem that the entire missing tty is written to the gui when you enable the tty gui...
Better also do this line in the false path

Reply:
Fixed.